### ISSUE TITLE
dist_deg : IndexError: list index out of range

### ISSUE THREAD

**Issue by ThomasCoutard** [ThomasParistech/pretty-gpx#1]
Hello, 
Thanks for your work !
Working on it help you to Dockerize it and to make it available online 🌍

I encountered a bug inside the algorithm (`IndexError`)
You'll find my GPX inside this archive to reproduce
[neouvieille.gpx.zip](https://github.com/user-attachments/files/16949593/neouvieille.gpx.zip)

<img width="750" alt="image" src="https://github.com/user-attachments/assets/5093a22b-a10b-4b53-b73d-df259b85c3bb">

Does your algorithm works on "special closed tracks" (talking about the path in North-East)

```
2024-09-10 20:51:19.701 | IndexError: list index out of range |  
-- | -- | --
  |   | 2024-09-10 20:51:19.701 | dist_deg = float(np.linalg.norm((self.list_lon[0] - self.list_lon[-1], |  
  |   | 2024-09-10 20:51:19.701 | File "/workspaces/pretty-gpx/pretty_gpx/gpx/gpx_track.py", line 63, in is_closed |  
  |   | 2024-09-10 20:51:19.701 | is_closed = gpx_track.is_closed(loose_ths_m) |  
  |   | 2024-09-10 20:51:19.701 | File "/workspaces/pretty-gpx/pretty_gpx/gpx/augmented_gpx_data.py", line 70, in from_path |  
  |   | 2024-09-10 20:51:19.701 | gpx_data = AugmentedGpxData.from_path(list_gpx_path) |  
  |   | 2024-09-10 20:51:19.701 | File "/workspaces/pretty-gpx/pretty_gpx/drawing/poster_image_cache.py", line 52, in from_gpx |  
  |   | 2024-09-10 20:51:19.701 | return PosterImageCaches.from_gpx(list_b, new_paper_size) |  
  |   | 2024-09-10 20:51:19.701 | File "/workspaces/pretty-gpx/pretty_gpx/main.py", line 34, in process_files |  
  |   | 2024-09-10 20:51:19.701 | return callback(*args, **kwargs) |  
  |   | 2024-09-10 20:51:19.701 | File "/usr/local/lib/python3.10/site-packages/nicegui/run.py", line 41, in safe_callback |  
  |   | 2024-09-10 20:51:19.701 | Traceback (most recent call last): |  
  |   | 2024-09-10 20:51:19.701 | Message: list index out of range |  
  |   | 2024-09-10 20:51:19.701 | Type: IndexError
```

Thanks

**Comment by ThomasParistech** [comment]
Hello!

It looks like your GPX doesn't contain any elevation data (the altitude is 0 everywhere). How did you capture this GPX?

I've manually recreated the same track on gpx.studio so that it contains the elevation, and it looks good :) 

I'll update the warnings to handle this case.

Thanks

![image](https://github.com/user-attachments/assets/53581288-1f41-4d73-987e-a5b348bed9cf)


Track with elevation data:
[corrected_neouvieille.zip](https://github.com/user-attachments/files/16958990/corrected_neouvieille.zip)

resulting poster :sparkles: 

![poster_Néouvielle_compressed](https://github.com/user-attachments/assets/b8ce4f34-e074-4fd6-9761-4cc39eb023b1)

**Comment by ThomasParistech** [comment]
@ThomasCoutard 
Did you really complete the entire hike in a single day? That's intense! Or did you camp midway? If so, consider splitting your GPX file in two and using the multi-day mode! 😊

**Comment by ThomasParistech** [comment]
Fixed in 7d442417479963b9f1d14156f01b126e6bd5d361
